fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneQueryTest.java

```diff
@@ -1271,4 +1277,55 @@ void continuations(boolean sorted) throws Exception {
             }
         }
     }
+
+    /**
+     * This tests is reproducing a thread deadlock that happened with Lucene threads when reading schema information.
+     * The fix for the deadlock has removed the {@code Map.computeIfAbsent} call (with the {@code synchronized} block)
+     * that blocked the {@code asyncToSync} calls.
+     * This test creates a large enough set of {@link org.apache.lucene.index.SegmentReader} futures so as to flood the thread pool,
+     * and then queries Lucene. THe large number of futures in the pool, together with the {@code synchronized} block
+     * would cause a deadlock.
+     */
+    @Test
+    void testQueryWithManyDocuments() {
+        String[] words = generateRandomWords(1000);
+        FDBDatabaseFactory.instance().setExecutor(new ForkJoinPool(Runtime.getRuntime().availableProcessors(),
+                ForkJoinPool.defaultForkJoinWorkerThreadFactory,
+                null, false));
```

Comment:
Setting it to `availableProcessors` might more accurately represent how someone would use a `ForkJoinPool` as the executor, but it makes this test less reproducible. For example if you ran this on a machine with 128 cores, it would probably pass even with the deadlock, but reproduce the issue on a machine with 4 cores (like a docker container, or a rather old computer).
4 might be too small and detect other deadlocks, so maybe 8 or 16 is more appropriate for now, although I think, ideally we would be able to use lucene with a million segments with a parallelism of 1.
```suggestion
        FDBDatabaseFactory.instance().setExecutor(new ForkJoinPool(4,
                ForkJoinPool.defaultForkJoinWorkerThreadFactory,
                null, false));
```

Reply:
Good point. The test should be reproducible under all conditions and the existing code would make it unpredictable.
Done.